### PR TITLE
Add select2 to layered nav

### LINKED ISSUE #13189: Implement SELECT2 for layered nav widget
This should be enabled when using dropdowns, and should support both OR and AND (or would use a multiselect, AND would be a regular select).

#11970

### PR DESCRIPTION
**Pull request by mikejolley**
Closes #13189 
Closes #13336 

Implements select2 for the layered nav if available for dropdowns.

Adds a button to submit multiple selects so multiple values can be chosen before it updates. 

backwards compatible with old queries.

### CODE REVIEW

File: includes/widgets/class-wc-widget-layered-nav.php [hunk 1]
```diff
@@ -213,16 +215,26 @@ protected function get_current_term_slug() {
 	 * @return bool Will nav display?
 	 */
 	protected function layered_nav_dropdown( $terms, $taxonomy, $query_type ) {
+		global $wp;
 		$found = false;
 
 		if ( $taxonomy !== $this->get_current_taxonomy() ) {
 			$term_counts          = $this->get_filtered_term_product_counts( wp_list_pluck( $terms, 'term_id' ), $taxonomy, $query_type );
-			$_chosen_attributes   = WC_Query::get_layered_nav_chosen_attributes();
+			$_chosen_attributes   = WC_Query                                                                                       ::get_layered_nav_chosen_attributes();
```
Comment (claudiosanches) [inline includes/widgets/class-wc-widget-layered-nav.php]: Missing `;`

Comment (claudiosanches) [inline includes/widgets/class-wc-widget-layered-nav.php]: Not sure if this even will work... Looks like that is missing more than just `;`.

**Review by claudiosanches — CHANGES_REQUESTED** [review]

File: includes/widgets/class-wc-widget-layered-nav.php (hunk 1 above)
Comment (mikejolley) [inline includes/widgets/class-wc-widget-layered-nav.php]: @claudiosanches no it's not, it's just be spaced weirdly.

Comment (claudiosanches) [inline includes/widgets/class-wc-widget-layered-nav.php]: Ohhhh, sorry xD

**Comment by claudiulodro** [comment]
I don't get any filters applying on the dropdown AND widget. It just loads up the page at `http://local.wordpress.dev/shop/?filter_color=`

**Comment by claudiulodro** [comment]
Additionally, there are some underlines on the regular widget that aren't present on the other layered nav widget:

screen shot 2017-07-11 at 8 40 50 am

**Comment by mikejolley** [comment]
@claudiulodro JS cache

**Comment by claudiulodro** [comment]
@mikejolley I'm busting it. All the JS is inline anyways, so I think it doesn't need busting. . .

I think the problem is you have a race condition with select2/Woo. On change, select2 updates the value in the hidden input. On change, we submit the form. The two race.

If I change it so that the form submits after 1 millisecond, everything works great. Feels like a hack to do it that way, though. I think if you do the 'change' listener on the hidden input, everything should also work properly.

**Pull request merged**